packages/squiggle-lang/src/rescript/Reducer/Reducer.resi

```diff
@@ -7,7 +7,20 @@ module MathJs = Reducer_MathJs
 
 @genType
 type expressionValue = ReducerInterface_ExpressionValue.expressionValue
-
+@genType
+type externalBindings = ReducerInterface_ExpressionValue.externalBindings
 @genType
 let evaluate: string => result<expressionValue, Reducer_ErrorValue.errorValue>
+@genType
+let evaluateWBindings: (
```

Comment:
maybe ``evaluateUsingExternalBindings``? 
(Note: I'm not a big fan of the ``W``; I assume it means ``with``, but am not sure)

Reply:
On that note, do we need this function? Can we just change ``evaluate`` to accept optional externalBindings?

Reply:
I hate the W also actually. But typing laziness  led to it actually.
I don't like the optional arguments for the sake of testing. This way it is clear that it needs its own tests and coverage.
And it will loose the clarity. Actually, maybe, that function need not be on the interface.  It's main purpose it to be able to test an expression with a collection of different bindings. And I guess you will need that also. 

evaluateWBindings is just evaluate with bindings. It has nothing to do with partial

Reply:
Renamed to evaluateUsingExternalBindings

Reply:
>  I don't like the optional arguments for the sake of testing. This way it is clear that it needs its own tests and coverage.

I don't think I agree here. I feel like we should get used to testing optional arguments, which definitely occur in different places. 

I believe it's the Rescript way to use optional arguments instead of one function per optional combination.